### PR TITLE
Implemented: notifications module for firebase notifications support

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/oms-api/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/modules/notification/index.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      params: payload
+    }) as any
+
+    if (!hasError(resp)) {
+      return Promise.resolve(resp.data.docs)
+    } else {
+      return Promise.reject(resp)
+    }
+  } catch (err) {
+    console.error(err)
```
Comment (ymaheshwari1) [inline src/modules/notification/index.ts:25]: Reject the promise in the catch block and in the else throw the resp.data

File: src/modules/notification/index.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  try {
+    const resp = await api({
+      url: "performFind",
+      method: "get",
+      params: payload
+    }) as any
+
+    if (!hasError(resp)) {
+      return Promise.resolve(resp.data.docs)
+    } else if (resp.data.error === 'No record found') {
```
Comment (ymaheshwari1) [inline src/modules/notification/index.ts:46]: The case of resp.data.error is already handled in `hasError` method.

File: src/modules/notification/index.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    console.error(err)
+  }
+}
+
+async function getNotificationUserPrefTypeIds(applicationId: string): Promise<any> {
+  const payload = {
+    "inputFields": {
+      "userPrefGroupTypeId": applicationId
+    },
+    "entityName": "UserPreference",
```
Comment (ymaheshwari1) [inline src/modules/notification/index.ts:34]: Add `fieldList` and `viewSize` param.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/modules/notification/index.ts (hunk 3 above)
Comment (k2maan) [reply src/modules/notification/index.ts:37]: Updated sir

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**